### PR TITLE
fix: add queue for Slack events

### LINKED ISSUE #1890: Get multiple responses for one slack input command
# Description
Using opsdroid with slack connector, there will be multiple responses for one slack input command.  From the debugging log, the same command is been detected and parsed for several times by opsdroid. 

After investigation, it seems the slack will retry if the server took longer than 3 seconds to respond to the previous event delivery attempt (https://api.slack.com/apis/connections/events-api#graceful_retries). Similar with the issue mentioned here https://github.com/slackapi/python-slack-sdk/issues/335

## Steps to Reproduce
Please also include relevant information and steps to reproduce the bug/issue.

## Expected Functionality
Expect opsdroid could provide a config setting for slack to acknowledge the slack API once the command is detected. 

## Experienced Functionality
Get multiple responses for one slack input command.

## Versions
- **Opsdroid version:** 0.24.1
- **Python version:**  3.8
- **OS/Docker version:** Redhat 8

## Configuration File
Please include your version of the configuration file below.

```yaml
# Your code goes here.

```

### PR DESCRIPTION
**Pull request by martinohmann**
# Description

The Slack Events API documentation states:

> Your app should respond to the event request with an HTTP 2xx within three seconds. If it does not, we'll consider the event delivery attempt failed. After a failure, we'll retry three times, backing off exponentially.
>
> [...]
>
> Respond to events with a HTTP 200 OK as soon as you can. Avoid actually processing and reacting to events within the same process. Implement a queue to handle inbound events after they are received.

https://api.slack.com/apis/connections/events-api#the-events-api__responding-to-events

This PR adds an `asyncio.Queue` which incoming events are pushed into. The `web_handler` will return HTTP Status 200 to Slack immediately and the event is processed asynchronously in the background.

I'm fairly new to async python, so please let me know if there is a more idiomatic way to achieve this and I'll update the PR accordingly.

We're already using this fix in production and so far it works flawlessly.

Fixes #1890

## Status
**READY**

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- Existing test suite and manual tests against a real Slack workspace.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: opsdroid/connector/slack/connector.py [hunk 1]
```diff
@@ -119,8 +134,13 @@ async def connect(self):
             _LOGGER.debug(_("Default room is %s."), self.default_target)
 
     async def disconnect(self):
-        """Disconnect from Slack. Only needed when socket_mode_client is used
-        as the Events API uses the aiohttp server"""
+        """Disconnect from Slack.
+
+        Cancels the event queue worker task and disconnects the
+        socket_mode_client if socket mode was enabled."""
```
Comment (FabioRosado) [inline opsdroid/connector/slack/connector.py:140]: Small nitpick, feel free to ignore
```suggestion
        socket_mode_client if socket mode was enabled.
        
        """
```

**Review by FabioRosado — COMMENTED** [review]
Thank you for working on this fix! Glad you have it working in production!
I'll test the fix later today and will approve later, I've also re-run CI to see if the windows tests pass

**Review by FabioRosado — APPROVED** [review]
Apologies took me longer than I wished for! Going to approve and merge this MR thank you for fixing the issue :smile: :+1:

**Comment by martinohmann** [comment]
@FabioRosado no worries, thanks for accepting the PR! :) Do you have any estimates when the next opsdroid release is planned? Thanks for your work!

**Comment by FabioRosado** [comment]
No worries! We tend to release monthly but we may skip a release if no new things got in.

I'll try to get some work done and maybe see if we can release soonish 😄

**Pull request merged**